### ISSUE TITLE
Rule proposal: no-duplicate-requires

### ISSUE THREAD

**Issue by pvdlg** [sindresorhus/eslint-plugin-unicorn#129]
[eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import) offer [a rule](https://github.com/benmosher/eslint-plugin-import/blob/master/docs/rules/no-duplicates.md) to report when a resolved path is imported more than once, but that doesn't work with `require`.

That would be great to have a similar feature for `require`.

### Fail
```js
const a = require('module');
const b = require('module');

const c = require('./file');
const d = require('./file');

const e = require('./file');
const f = require('././file');
```

**Comment by Manu1400** [comment]
My first ESLint plugin with the rule _no-duplicate-requires_: https://github.com/Manu1400/eslint-plugin-emmanuel

**Comment by sindresorhus** [comment]
I'm gonna pass on this. I don't want to maintain code that only targets `require` as it's no legacy that will be moot in the future. `import` is the future, for better or worse.
